module/solutions/helper.js

```diff
@@ -1967,6 +1968,67 @@ module.exports = class SolutionsHelper {
     });
   }
 
+  /**
+   * Solution Report Information.
+   * @method
+   * @name read
+   * @param {String} solutionId - Solution Id.
+   * @param {String} userId - User Id.
+   * @returns {Object} - Report Information of the solution.
+   */
+
+  static read(solutionId, userId) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let userInformation = await userExtensionsHelper.userExtensionDocument({
+            userId: userId,
+            "programRoles.code" : { $in : ["PROGRAM_MANAGER","PROGRAM_DESIGNER"]},
+            status: constants.common.ACTIVE,
+            isDeleted: false
+        }, { _id: 1 });
+
+        if (!userInformation) {
```

Comment:
@priyanka-TL we should also check whether or not he is program manager or designer of this particular solution